stream-lava-log.py

```diff
@@ -19,76 +19,192 @@
+        return self.server
+
+
+
+class ArgumentParser(object):
+    def __init__(self, args):
+        self.username = args.get('username')
+        self.token = args.get('token')
+        self.server = args.get('server')
+        self.job = args.get('job')
+
+    def get_username(self):
+        return self.username
+
+    def get_token(self):
+        return self.token
+
+    def get_server(self):
+        return self.server
+
+    def get_job(self):
+        return self.job
+
+
+class LavaConnection(object):
+    def __init__(self, configuration):
+        self.configuration = configuration
+        self.connection = None
+
+    def connect(self):
+        url = self.configuration.construct_url()
         try:
-            current_job_file = str(connection.scheduler.job_output(args.job))
-            if old_job_file is not None:
-                latest = current_job_file[len(old_job_file):]
-            else:
-                latest = current_job_file
-            for s in stream_string(latest):
-                print s
-            time.sleep(2)
-            old_job_file = current_job_file
-            status = connection.scheduler.job_status(args.job)
-            if status['job_status'] in end_states:
-                print 'Job has finished'
-                run = False
-        except (xmlrpclib.ProtocolError, xmlrpclib.Fault, IOError):
-            pass
+            print "Connecting to Server..."
+            self.connection = xmlrpclib.ServerProxy(url)
+
+            print "Connection Successful!"
+            print "connect-to-server : pass"
+        except (xmlrpclib.ProtocolError, xmlrpclib.Fault, IOError) as e:
+            print "CONNECTION ERROR!"
+            print "Unable to connect to %s" % url
+            print e
+            print "connect-to-server : fail"
+            exit(1)
+
+    def get_job_status(self, job_id):
+        return self.connection.scheduler.job_status(job_id)
+
+    def get_job_output(self, job_id):
+        return self.connection.scheduler.job_output(job_id)
+
+
+class LavaRunJob(object):
+    def __init__(self, connection, job_id):
+        self.END_STATES = ['Complete', 'Incomplete', 'Canceled']
+        self.job_id = job_id
+        self.printed_output = None
+        self.connection = connection
+
+    def get_status(self):
+        return self.connection.get_job_status(self.job_id)['job_status']
+
+    def get_output(self):
+        return self.connection.get_job_output(self.job_id)
+
+    def is_running(self):
+        return self.get_status() not in self.END_STATES
+
+    def print_output(self):
+        full_output = str(self.get_output())
+        if self.printed_output:
+            new_output = full_output[len(self.printed_output):]
+        else:
+            new_output = full_output
+        print new_output,
+        self.printed_output = full_output
+
+    def run(self):
+        is_running = True
+
+        self.connection.connect()
```

Comment:
Be careful here. The username, token and server are not validated at this point. I usually make a test API call to ensure the server connection is alive.

[test]
server: http://lava/RPC2
username: tyler
token: nope